src/app/reporting/SynchronizedReportSchedulerImpl.h

```diff
@@ -30,6 +30,48 @@ using Milliseconds64  = System::Clock::Milliseconds64;
 using ReadHandlerNode = ReportScheduler::ReadHandlerNode;
 using TimerDelegate   = ReportScheduler::TimerDelegate;
 
+/**
+ * @class Synchronized ReportSchedulerImpl
+ *
+ * @brief This class extends ReportSchedulerImpl and overrides it's scheduling logic.
+ *
+ * It only overrides Observers method where the scheduling logic make it necessary, the others are kept as is.
```

Comment:
I am confused about what "Observers method" is... One method, ok... but what are the plural "Observers" and how is the method related to them?